sigstore/_internal/rekor/__init__.py

```diff
@@ -30,6 +31,26 @@
 __all__ = ["RekorClient", "SignedCheckpoint"]
 
 
+def _dsse_from_parts(cert: Certificate, evp: dsse.Envelope) -> rekor_types.Dsse:
+    signature = rekor_types.dsse.Signature(
+        signature=evp._inner.signatures[0].sig,
```

Comment:
It's an assumption for now, although my understanding is that the same assumption is made/asserted as an invariant by Rekor (i.e., a submitted DSSE entry can only have one signature).

CC @haydentherapper for confirmation 🙂

Reply:
Rekor allows DSSEs with multiple signatures - https://github.com/sigstore/rekor/blob/main/pkg/types/dsse/v0.0.1/entry.go#L371-L373. intoto v0.0.1 had this [limitation](https://github.com/sigstore/rekor/blob/main/pkg/types/intoto/v0.0.1/entry.go#L230-L231) for this was fixed in intoto v0.0.2 and the dsse kind.

Reply:
Gotcha, maybe I should make this more flexible, then --  is my read correct that the logic here is "all signatures must verify for the entry to be accepted"?

Reply:
Correct!

Reply:
Alright, I've done some tweaking of this in https://github.com/sigstore/sigstore-python/pull/962/commits/e1753ffea3c3376068c13ded0375602f38c993e5 -- we now handle multiple signatures gracefully when they appear in the `Envelope`, although I suspect this will never happen in practice in the Sigstore setting (since the bundle only supports a single signing cert?)